### PR TITLE
mesa 22.0.5

### PR DESCRIPTION
**Pull request by yurikoles**
Created with `brew bump-formula-pr`.

~~`resource` blocks may require updates.~~

### CODE REVIEW

**Comment by chenrui333** [comment]
known linux failure

`brew test --retry --verbose pdfsandwich`

**Review by chenrui333 — DISMISSED** [review]

File: Formula/mesa.rb [hunk 1]
```diff
@@ -27,10 +27,11 @@ class Mesa < Formula
   depends_on "libxcb"
   depends_on "libxdamage"
   depends_on "libxext"
+  depends_on "llvm"
```
Comment (carlocab) [inline Formula/mesa.rb:30]: Why is this needed? A dependency change on a patch release seems quite unlikely.

**Review by carlocab — CHANGES_REQUESTED** [review]

File: Formula/mesa.rb (hunk 1 above)
Comment (yurikoles) [reply Formula/mesa.rb:30]: It's not a hard dependency, but rather a delayed fix. I see more targets are being built with brew `llvm`, i.e. `mesa` doesn't like stock `llvm` for some reason.

Comment (yurikoles) [reply Formula/mesa.rb:30]: And `homebrew/core` discourage `recommended` and `optional` dependencies.

Comment (carlocab) [reply Formula/mesa.rb:30]: That seems to apply only on Linux and not macOS. What kind of issues do we see from not having Homebrew LLVM as a dependency?

Comment (yurikoles) [reply Formula/mesa.rb]: I tried to collect facts, but the only difference is that `LLVM` is being "found", ≈100 extra steps in `ninja`, but files count in resulting bottle is same, and they have mostly same sizes, so I moved `llvm` to `on_linux` with comment.

File: Formula/mesa.rb [hunk 2]
```diff
@@ -47,6 +47,7 @@ class Mesa < Formula
     depends_on "libxv"
     depends_on "libxvmc"
     depends_on "libxxf86vm"
+    depends_on "llvm" # can't use from macOS, it isn't found
```
Comment (carlocab) [inline Formula/mesa.rb:50]: That doesn't make sense.
```ruby
uses_from_macos "foo"
```
is the exact same thing as
```ruby
on_linux do
  depends_on "foo"
end
```

Comment (yurikoles) [reply Formula/mesa.rb:50]: `uses_from_macos "llvm"` is a false statement, it isn't found and isn't used.

Comment (carlocab) [reply Formula/mesa.rb:50]: `uses_from_macos "llvm"` is a no-op on macOS (because of the equivalence above), so it being false doesn't really make sense.

Comment (yurikoles) [reply Formula/mesa.rb:50]: It may be equivalent in case formula actually detects and uses stock `llvm`, it's not the case for `mesa`, so I wanted to be more correct in stating `llvm` dependency in only `on_linux`.

Comment (SMillerDev) [reply Formula/mesa.rb:50]: So what does it use on macOS?

Comment (yurikoles) [reply Formula/mesa.rb:50]: It uses stock `llvm`, but only as any other compiler. `mesa` is tied with `llvm` on other platforms, where real drivers work, e.g. `llvm` has backend for Radeon GPUs.

Comment (SMillerDev) [reply Formula/mesa.rb:50]: I don't get it, if it uses `llvm` on macOS the `uses_from_macos "llvm"` is correct right?

Comment (yurikoles) [reply Formula/mesa.rb:50]: AFAIK, we don't set `uses_from_macos "llvm"` for all formula that just needs any C compiler.

Comment (yurikoles) [reply Formula/mesa.rb:50]: I mean that `mesa` may be compiled with any modern compiler and it doesn't use any LLVM-specific features macOS, because it doesn't recognize it as suitable LLVM instance in case of stock macOS.
And in case of advertising brewed LLVM it detects and "uses" it, but the resulting product is the same.

Comment (yurikoles) [reply Formula/mesa.rb:50]: So what is the solution? It doesn't need LLVM on macOS, only on Linux, so let's keep current code?

Comment (yurikoles) [reply Formula/mesa.rb]: Done

**Review by SMillerDev — DISMISSED** [review]

File: Formula/mesa.rb [hunk 3]
```diff
@@ -91,30 +91,28 @@ def install
...
-        args << "-Dgles2=enabled"
-        args << "-Dgallium-xvmc=disabled"
-        args << "-Dvalgrind=false"
-        args << "-Dtools=drm-shim,etnaviv,freedreno,glsl,nir,nouveau,xvmc,lima"
-      end
-
-      system "meson", *std_meson_args, "..", *args
-      system "ninja"
-      system "ninja", "install"
+    args = std_meson_args + ["-Db_ndebug=true"]
```
Comment (carlocab) [inline Formula/mesa.rb:94]: Not a fan of putting `std_meson_args` here, personally. Better to just include it separately in the call to `meson`.

**Review by carlocab — DISMISSED** [review]

File: Formula/mesa.rb (hunk 3 above)
Comment (yurikoles) [reply Formula/mesa.rb:94]: I don't mind about that fix, but it will trigger another 4.5h run on Linux. I promise to address it in #101859.

**Review by SMillerDev — APPROVED** [review]
